docs/reference/modules/cluster/remote-clusters-api-key.asciidoc

```diff
@@ -0,0 +1,177 @@
+[[remote-clusters-api-key]]
+=== Add remote clusters using API key authentication
+
+coming::[8.10]
+beta::[]
+
+API key authentication enables a local cluster to authenticate itself with a
+remote cluster via a cross-cluster API key. The API key needs to be created by
+an administrator of the remote cluster. The local cluster is configured to
+provide this API key on each request to the remote cluster. The remote cluster
+verifies the API key and grants access, based on the API key's privileges.
+
+All cross-cluster requests from the local cluster are bound by the API key's
+privileges, regardless of local users associated with the requests. For example,
+if the API key only allows read access to `my-index` on the remote cluster, even
+a superuser from the local cluster is limited by this constraint. This mechanism
+enables the remote cluster's administrator to have full control over who can
+access what data with cross-cluster search and/or cross-cluster replication. The
+remote cluster's administrator can be confident that no access is possible
+beyond what is explicitly assigned to the API key.
+
+On the local cluster side, not every local user needs to access every piece of
+data allowed by the API key. An administrator of the local cluster can further
+configure additional permission constraints on local users so each user only
+gets access to the necessary remote data. Note it is only possible to further
+reduce the permissions allowed by the API key for individual local users. It is
+impossible to increase the permissions to go beyond what is allowed by the API
+key.
+
+In this model, cross-cluster operations use a dedicated server port (remote
+cluster interface) for communication between clusters. A remote cluster must
+enable this port for local clusters to connect. Configure Transport Layer
+Security (TLS) for this port to maximize security. 
```

Comment:
Should we somehow indicate that the below section is going to show how to configure TLS?